### DISCUSSION TITLE
Managing and Prioritizing Multiple Audio Channels with ESP32

### DISCUSSION THREAD

**Discussion by theSchoofseggl** [pschatzmann/arduino-audio-tools#1684] [Q&A] [answered]
Grüezi and Hello everyone,

I recently came across [this discussion](https://github.com/pschatzmann/arduino-audio-tools/discussions/694) where someone faced a similar challenge to what I'm currently working on. 

I’m working on a project where I need to manage and prioritize three audio channels from three different radios. While high-definition audio isn’t necessary, it’s crucial that the channels are sampled and prioritized correctly. The highest-priority channel should then be output via Bluetooth, essentially creating a multiplexer.

To manage multiple channels, I’m considering using three analog Schmitt triggers to monitor the signals and send an "Audio ON" signal to a digital pin. A digital output would then switch the three analog signals to an I2C ADC. The software would handle the switching between the signals, similar to this project: [Audio Switcher Arduino](https://www.instructables.com/Audio-Switcher-Arduino/).

I'm curious if anyone has tackled a similar problem and how it was ultimately solved. Any insights or advice would be greatly appreciated! Any thoughts on this Project?

Thanks in advance!

Best regards,
Philipp

**Comment by pschatzmann** [comment]
Are your radios mono or stereo ?

**Reply by theSchoofseggl** [reply]
The Radios are old analog devices, they are mono.

**Comment by pschatzmann** [comment, marked as answer]
The ESP32 has 2 I2S ports, so you can use 2 I2S ADCs. I2S is stereo, so you get max 2 * 2  = 4 audio input channels.

Therfore you have also the option to use 2 I2S ADCs and do your multiplexing in software

**Reply by theSchoofseggl** [reply]
Ahh, then I understood the discussion above wrong. You normally normally recommended the blue i2s adc with the wm8782s?

**Reply by pschatzmann** [reply]
So far I have only tested [one ADC that can be found here](https://github.com/pschatzmann/arduino-audio-tools/wiki/External-ADC)

**Reply by theSchoofseggl** [reply]
Thats what I mean, I think the I2S ADC Audio I2S Capture Card Module use the wm8782s. But I'm not 100% sure. Thank you for your help!

**Comment by pschatzmann** [comment]
For a hardware based analog multiplexer solution: I found [this cheap module](https://www.aliexpress.com/item/1005006813298019.html)

You could use one to select the input to the audio ADC and one to monitor the volume on each line with the internal ADC which then decides on the input to select.

**Reply by theSchoofseggl** [reply]
That sounds also like a smart solution. I think it reduce the computational effort for the uC (what makes him faster).

The Downside is more soldering work. But I have to solder [differential amplifier](https://www.amazon.de/CJMCU-333-Multifunktionales-3-Operationsverst%C3%A4rkermodul-Pr%C3%A4zisionsinstrumentenverst%C3%A4rker-OPAMP-Modul/dp/B091GS4P24) anyway, since a radio can only be measured differentially, so it wouldn't increase the effort much. 

What solution would you prefer?
